### PR TITLE
parser: Return empty if there is no tokens

### PR DESCRIPTION
**Pull request by qinqon**
If a user pass an empty expression parser do an illegal access to the
tokens list, this change add a test and a fix to mimic this scenario.

### CODE REVIEW

**Review by AlonaKaplan — DISMISSED** [review]

File: nmpolicy/internal/parser/parser_test.go [hunk 1]
```diff
@@ -287,6 +293,7 @@ func fromTokens(tokens ...lexer.Token) test {
 		t.tokens[i].Position = len(t.expression)
 		t.expression += t.tokens[i].Literal
 	}
+	fmt.Println(t.expression)
 	return t
```
Comment (AlonaKaplan) [inline nmpolicy/internal/parser/parser_test.go:297]: Why do you print to strout?

File: nmpolicy/internal/parser/parser_test.go [hunk 2]
```diff
@@ -36,6 +36,12 @@ func TestParser(t *testing.T) {
 
 func testParseFailures(t *testing.T) {
 	var tests = []test{
+		{
```
Comment (AlonaKaplan) [inline nmpolicy/internal/parser/parser_test.go:39]: Why did you add the test to `testParseFailures`? I doesn't fail, just returns am empty ast.
BTW, I think we should fail and not allow an empty expression. But I guess now the failure will happen in the resolver, right?

**Review by AlonaKaplan — COMMENTED** [review]
Approved too fast. Two small comments.

File: nmpolicy/internal/parser/parser_test.go [hunk 3]
```diff
@@ -36,6 +36,12 @@ func TestParser(t *testing.T) {
...
+			tokens: []lexer.Token{},
+			expected: expected{
+				ast: &ast.Node{},
+			},
+		},
```
Comment (EdDev) [inline nmpolicy/internal/parser/parser_test.go:44]: Don't you need to give a description to what is tested here?
And I would wrap it with a `expectSuccess`, similar to what you did in the errors ones.

**Review by EdDev — COMMENTED** [review]
> If a user pass an empty expression parser do an illegal access to the
tokens list, this change add a test and a fix to mimic this scenario.

The `do` word seems not a fit. Can you rephrase/fix this sentence?
Also, the "this change..." should be a new sentence.

File: nmpolicy/internal/parser/parser_test.go (hunk 1 above)
Comment (qinqon) [reply nmpolicy/internal/parser/parser_test.go:297]: My bad fixing.

File: nmpolicy/internal/parser/parser_test.go (hunk 2 above)
Comment (qinqon) [reply nmpolicy/internal/parser/parser_test.go]: Right, moving to Success, yes.

File: nmpolicy/internal/parser/parser_test.go (hunk 3 above)
Comment (qinqon) [reply nmpolicy/internal/parser/parser_test.go]: Description is composed from the expression, https://github.com/nmstate/nmpolicy/blob/main/nmpolicy/internal/parser/parser_test.go#L277-L282.

Adding a expectSuccess with empty stuff.

**Review by EdDev — APPROVED** [review]
